core/src/main/java/org/apache/gravitino/storage/relational/service/TableMetaService.java

```diff
@@ -229,6 +231,31 @@ public boolean deleteTable(NameIdentifier identifier) {
           if (deleteResult.get() > 0) {
             TableColumnMetaService.getInstance().deleteColumnsByTableId(tableId);
           }
+        },
+        () -> {
+          if (deleteResult.get() > 0) {
+            SessionUtils.doWithoutCommit(
+                SecurableObjectMapper.class,
+                mapper ->
+                    mapper.softDeleteObjectRelsByMetadataObject(
+                        tableId, MetadataObject.Type.TABLE.name()));
+          }
+        },
+        () -> {
+          if (deleteResult.get() > 0) {
+            SessionUtils.doWithoutCommit(
+                TagMetadataObjectRelMapper.class,
+                mapper ->
+                    mapper.softDeleteTagMetadataObjectRelsByMetadataObject(
+                        tableId, MetadataObject.Type.TABLE.name()));
+          }
+        },
+        () -> {
+          if (deleteResult.get() > 0) {
+            SessionUtils.doWithoutCommit(
+                TagMetadataObjectRelMapper.class,
+                mapper -> mapper.softDeleteTagMetadataObjectRelsByTableId(tableId));
+          }
```

Comment:
Can we merge this operation into one like 

if (deleteResult.get() > 0) {
    
   // one
   SessionUtils.doWithoutCommit()
  
  // two
  SessionUtils.doWithoutCommit()
 
  ...
}

Reply:
OK.